CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# Contributing to the Eclipse Che4z LSP for HLASM project
+
+Thanks for your interest in this project.
+
+## Eclipse Che4z
+
+This project is a part of [Eclipse Che4z](https://projects.eclipse.org/projects/ecd.che.che4z) which provides components/extensions for Eclipse Che to facilitate mainframe application development.
+
+## Eclipse Contributor Agreement
+
+Before your contribution can be accepted by the project team contributors must electronically sign the Eclipse Contributor Agreement (ECA).
+
+* http://www.eclipse.org/legal/ECA.php
+
+Commits that are provided by non-committers must have a Signed-off-by field in the footer indicating that the author is aware of the terms by which the contribution has been provided to the project. The non-committer must additionally have an Eclipse Foundation account and must have a signed Eclipse Contributor Agreement (ECA) on file.
+
+Make sure that the user email you specified on your local git is the same as on the Github account.
+
+For more information, please see the Eclipse Committer Handbook:
+https://www.eclipse.org/projects/handbook/#resources-commit
+
+Such signoff is easily achieved using the `--signoff` option of the `git commit` command (provided that the git credentials are properly configured):
+```
+git commit --signoff
+```
+
+## Setting up the Development Environment
+
+The project uses CMake targeting all major platforms (Windows, Linux, MacOS). The full build instructions including project prerequisities can be found in the project [wiki](https://github.com/eclipse/che-che4z-lsp-for-hlasm/wiki/Build-instructions).
+
+## Start contributing
+
+Before contributing to the project, fork the repository and clone it, or add it as a new remote if you have already cloned the original repository. All the commits should be pushed to the fork in order to open pull requests from it.
+
+We encourage every potential contributor to read the project documetation available on the project [wiki](https://github.com/eclipse/che-che4z-lsp-for-hlasm/wiki/). It describes the architecture of the project, decomposes it into software components and explains their relationship.
+
+All the activity on the project should begin with defining a new issue that describes the required changes.
+
+## Contribution acceptance
+
+Each pull request is automatically verified in the following environments (in a GitHub Actions pipeline):
+
+- Windows build with the MSVC compiler
```

Comment:
Shouldn't we specify which MSVC?

Reply:
truth is, the github updates it periodically so we dont really have control over that.